module/hotspot/hsperfdata/hsperfdata.go

```diff
@@ -89,59 +162,67 @@ func (m *MetricSet) buildMapStr(entries []PerfDataEntry) common.MapStr {
 	return event
 }
 
-func (m *MetricSet) publishAll() (common.MapStr, error) {
-	f, err := os.Open(m.hsPerfDataPath)
+func (p *ProcStats) publishAll() (common.MapStr, error) {
+	f, err := os.Open(p.hsPerfDataPath)
 	if err != nil {
 		return nil, err
```

Comment:
If HSBeat tries to open hsperfdata which is owned by other user, it is failed at this point.
I agree HSBeat should scan all process on the machine. But we should treat other user process.